src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs

```diff
@@ -50,7 +82,16 @@ public override async Task OnConnectedAsync(HubConnectionContext connection)
                 try
                 {
                     InitializeHub(hub, connection);
-                    await hub.OnConnectedAsync();
+
+                    if (_onConnectedMiddleware != null)
```

Comment:
Is this to save us allocating the context objects when there are no filters?

Reply:
Yeah